sklearn/tree/_tree.pyx

```diff
@@ -178,19 +194,17 @@ cdef class DepthFirstTreeBuilder(TreeBuilder):
         cdef SIZE_t max_depth_seen = -1
         cdef int rc = 0
 
-        cdef Stack stack = Stack(INITIAL_STACK_SIZE)
+        cdef stack[StackRecord] builder_stack
         cdef StackRecord stack_record
 
         with nogil:
             # push root node onto stack
-            rc = stack.push(0, n_node_samples, 0, _TREE_UNDEFINED, 0, INFINITY, 0)
-            if rc == -1:
-                # got return code -1 - out-of-memory
-                with gil:
-                    raise MemoryError()
+            builder_stack.push({"start": 0, "end": n_node_samples, "depth": 0,
+                                "parent": _TREE_UNDEFINED, "is_left": 0,
+                                "impurity": INFINITY, "n_constant_features": 0})
```

Comment:
There is no python interaction here as it is in the `nogil` scope. The C++ code generated from this `push` sets the struct and pushes it onto the stack. (This was kind of surprising to me)

Reply:
It's nice that Cython proposes this syntactic sugar.

Reply:
Some magical Cython stuff :)

Reply:
(could we make one line by entry even if it is Cython code)